Dockerfile

```diff
@@ -112,9 +126,28 @@ RUN apt-get update && apt-get install -y \
         tmux \
         unzip \
         vim \
-        x-window-system \
-        ros-foxy-rmw-cyclonedds-cpp
+        x-window-system"
+
+RUN apt-get update && apt-get install -y lsb-release && apt-get clean ALL
 
+# Install ROS Noetic
+ARG ROS_DISTRO=noetic
+RUN sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list' && \
+    apt-key adv --keyserver 'hkp://keyserver.ubuntu.com:80' --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654 && \ 
+    apt-get update && \
+    apt-get install --yes ${BASE_DEPS}
+
+# Prepare for ROS 2 Foxy installation
+RUN apt update && apt install locales \
```

Comment:
Are the apt updates here and line 145 necessary? I don't see that anything would change the apt sources/package list until after the apt-key add at the end of  lines 146-147.

Reply:
While moving the `&&` from the beginning of the line to the end is not necessary for this exact change, the git diff is going to show those lines as changed anyway due to the introduction of the variable lines so I went ahead and reformatted them to be consistent with the `&&` placement throughout the rest of the Dockerfile.

I can revert them if you like but the diff is still going to think they are changed.

Reply:
I'm less concerned about the ordering and more just wondering if the actual `apt-get update` step is necessary in the first place (unless that's you meant). As far as I can tell there's no need for these, and apt-get update is one of the more time consuming steps in the build, so removing 1 or 2 that aren't needed can be a big win.

If you mean that these are actually being removed in a later change set though, then that's totally fine though to keep this PR focused and this change can be addressed later.

Reply:
Yup, I will be refactoring everything in PRs to follow, I'm breaking it up for review-ability.